qlib/log.py

```diff
@@ -29,6 +31,15 @@ def __init__(self, module_name):
         self.module_name = module_name
         self.level = 0
 
+    def __getstate__(self):
```

Comment:
Why  do you have to implement these three magic functions? 
Aren't they the default behaviors of an Python Object?

Reply:
Maybe these functions should also be skipped in the  `MetaLogger.__new__` function.

Reply:
It is necessary to add these two methods in the class. These functions are used by pickle to customized how objects are pickled. If we do not include these in the class, the `__getattr__` method will be called while pickling and results in recursion error.
![Screenshot from 2021-04-30 09-31-04](https://user-images.githubusercontent.com/26081991/116637722-45bb7100-a997-11eb-9998-c5e960385294.png)

Besides the original dict did not have these two methods either.
![Screenshot from 2021-04-30 09-30-44](https://user-images.githubusercontent.com/26081991/116637766-61267c00-a997-11eb-949e-acc94f77d181.png)

It is necessary to specify these two methods if we want to pickle the class.